feat: add lark auth support

mashizora opened the pull request:
Lark is a productivity app for teams, just like MS Teams.

This PR add Lark as an oauth2 provider.

mashizora committed:
add lark oauth support

mashizora committed:
fix oauth scopes to support token refresh

ganigeorgiev commented on tools/auth/lark.go:
```diff
@@ -0,0 +1,82 @@
+	user := &AuthUser{
+		Id:           extracted.Data.Id,
+		Name:         extracted.Data.Name,
+		Email:        extracted.Data.Email,
```
There is a warning at the beginning of https://open.feishu.cn/document/server-docs/authentication-management/login-state-management/get:

> Phone numbers and email addresses are contact details imported by the organization administrator and have not been verified by the users in real time. **Developers are advised not to use them directly as login credentials for business systems**. If used, make sure to verify them independently.

PocketBase requires the email address loaded in `AuthUser` to be always verified and since they are not verified by Lark we can't trust it and we should skip the `Email` field assignment.

ganigeorgiev reviewed (changes requested):
Thank you. I'll try to review and test the provider in more details sometime this weekend but for now I left some minor things that you can check when you find time.

mashizora committed:
skip email field assignment and some minor fixes

mashizora commented on tools/auth/lark.go:
```diff
@@ -0,0 +1,82 @@
+
+	extracted := struct {
+		Data struct {
+			Id        string `json:"user_id"`
```
Your worries do exist because Lark does not provide a real global user id, and there is no document about weather two different users in different tenants could have the different `user_id`. The reason might be some privacy or security problems or whatever.

My company use Lark and I can try to explain something to help you making the final decision.

OAuth2 apps in Lark has two types, marketplace apps and enterprise in-house apps. 

- For in-house apps, `user_id` has no problems because all operations are under the same tenant. 
- When it comes to marketplace apps, things get to complicated. We can use `union_id` to avoid `user_id` conflict, or make a new id based on tenant and `user_id` (I don't think this is a good choice).

The cons of `union_id` is, if we change the OAuth2 app config in PocketBase to another app, which is published by another Lark developer (our dashboard supports admin to do this), the all links might be broken. Maybe we can change to use `union_id` and leave a warning message in the frontend.

mashizora committed:
switch to use union id as the pocketbase user id

refs:
- https://open.feishu.cn/document/platform-overveiw/basic-concepts/user-identity-introduction/introduction
- https://open.feishu.cn/document/platform-overveiw/basic-concepts/user-identity-introduction/user-id

ganigeorgiev commented:
Thank you for the latest commit changes and information about the different ids.

I've spent a little bit more time with the Lark documentation and I think using the `union_id` as in your last commit is the better and safer option. It also doesn't seem to require special scopes and it is the closest to a global ID that we have right now _(per one of the links you've shared there is something called `lark_id` but I guess it is not exposed and used only for internal purposes)_.

To avoid confusion and eventual misuse, I've added a note in the UI explaining that we are using the union id.

I've merged the provider for now in the `develop` branch and it will be available with the next feature release (most likely tomorrow or next weekend).
